fossunited/ticketing/doctype/foss_event_ticket/test_foss_event_ticket.py

```diff
@@ -78,3 +88,59 @@ def test_checkin_as_non_chapter_member(self):
         # Then the user should not have permission to check-in the attendee
         with self.assertRaises(frappe.PermissionError):
             checkin_attendee(self.event.name, ticket.as_dict(), "test2@example.com")
+
+    def test_create_when_tickets_closed(self):
+        # Given a paid event with ticket status as "Closed"
+        self.event.tickets_status = "Closed"
+        self.event.save()
+        # When a ticket is created for this event
+        # Then an error must be thrown
+        with self.assertRaises(frappe.PermissionError):
+            insert_test_ticket(event=self.event.name)
+
+    def test_ticket_creation(self):
+        # Given an event
+        # When the tickets_status is Live
+        self.event.tickets_status = "Live"
+        self.event.save()
+
+        # Then a ticket for this event should be created without any problem
+        ticket = insert_test_ticket(event=self.event.name)
+        ticket.delete(force=True)
+
+    def test_ticket_tier_closing(self):
+        # Given an event with a ticket tier
+        tier = self.event.get("tiers")[0]
+
+        # Make sure this tier is enabled
+        self.assertTrue(tier.enabled)
+
+        # make sure there are 0 tickets of this tier to start with
+        self.assertEqual(
+            frappe.db.count(
+                EVENT_TICKET,
+                {
+                    "event": self.event.name,
+                    "tier": tier.title,
+                },
+            ),
+            0,
+        )
+
+        # When maximum number of tickets are created of this tier type
+        tickets = []
+        for _ in range(tier.maximum_tickets):
+            _ticket = insert_test_ticket(
+                event=self.event.name,
+                tier=tier.title,
+            )
+            tickets.append(_ticket)
+
+        self.event.reload()
```

Comment:
You are testing the wrong thing here 😅

Reply:
Insert in bulk?

Reply:
Not using bulk insert since I want to use the custom util function which handles the field values for the doctype